test/src/offline_transport/internet_connectivity_service_test.dart

```diff
@@ -19,12 +20,14 @@ void main() {
   late List<ConnectivityResult> fakeConnectivityResults;
   late StreamController<List<ConnectivityResult>> fakeConnectivityController;
   late InternetConnectivityService service;
+  late NetworkReachabilityChecker networkReachabilityChecker;
 
   setUp(() {
     mockConnectivity = MockConnectivityChecker();
     fakeConnectivityController =
         StreamController<List<ConnectivityResult>>.broadcast();
     fakeConnectivityResults = [ConnectivityResult.none];
+    networkReachabilityChecker = NetworkReachabilityCheckerFactory().create();
```

Comment:
Instead of using the real `networkReachabilityChecker` here, what do you think about if we mock it, similar to how we mock the `ConnectivityChecker`?

we could do something like

```
class MockNetworkReachabilityChecker extends Mock implements NetworkReachabilityChecker {}
...
when(() => mockNetworkReachabilityChecker. isConnectedToInternet(any))
         .thenAnswer((_) async => fakeIsConnected);
```